x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -204,7 +204,7 @@ private static Object innerDeepCopy(Object value, boolean unmodifiable) {
      * should thus be included in the cache key
      */
     // Visibility for testing
-    record CacheKey(String enrichIndex, Object lookupValue, int maxMatches) {
+    record CacheKey(String enrichIndex, ProjectId projectId, Object lookupValue, int maxMatches) {
```

Comment:
Tiniest nit ever: intuitively, I would have put the project ID as the first parameter because it has the largest "scope". Maybe that's just a weird personal preference I have, so this is definitely optional.